src/renderer/src/hooks/stores/mapeoDeviceStore.ts

```diff
@@ -0,0 +1,103 @@
+import { create } from 'zustand'
+
+type DeviceType = 'mobile' | 'desktop'
+
+export type Role = 'coordinator' | 'participant'
+
+type Device = {
+  name: string
+  deviceId: string
+  deviceType: DeviceType
+  pastDeviceNames?: string[]
+}
+
+type MemberDevice = {
+  role: Role
+  dateAdded: Date
+} & Device
+
+type MapeoDeviceState = {
+  nonMemberDevices: Record<Device['deviceId'], Device>
+  memberDevices: Record<MemberDevice['deviceId'], MemberDevice>
+  actions: {
+    addDeviceToProject: (deviceId: string, role: Role) => void
+    removeDeviceFromProject: (deviceId: string) => void
+    changeDeviceRole: (deviceId: string, newRole: Role) => void
+  }
+}
+
+export const useMapeoDeviceStore = create<MapeoDeviceState>()((set) => ({
+  nonMemberDevices: createRandomDevices(),
+  memberDevices: {},
+  actions: {
+    addDeviceToProject: (deviceId, role) =>
+      set((state) => {
+        if (state.memberDevices[deviceId]) {
+          throw new Error(`Device is already part of the project`)
+        }
+
+        const deviceToBeAdded = state.nonMemberDevices[deviceId]
+
+        if (!deviceToBeAdded) {
+          throw new Error(`Device does not exist`)
+        }
+
+        const { [deviceId]: key, ...rest } = state.nonMemberDevices
```

Comment:
Very nonblocking suggestion:

```suggestion
        const { [deviceId]: deviceToBeAdded, ...newNonMemberDevices }  = state.nonMemberDevices

        if (!deviceToBeAdded) {
          throw new Error(`Device does not exist`)
        }
```

Reply:
https://github.com/digidem/mapeo-desktop-shell/pull/40/commits/9d76998e7e969ecdb734c309a9249eed7a8b74c6